solarnet/README.md

```diff
@@ -93,4 +123,73 @@ one-liners be automated.
 
 ### unsupported parameter for module: restart_policy
 
-You're probably using the system version of Ansible, and it is outdated (< 1.9). Make sure to follow the Ansible setup steps above, and load the virtualenv. This will load a working version of Ansible.
+You're probably using the system version of Ansible, and it is outdated (< 1.9).
+Make sure to follow the Ansible setup steps above, and load the virtualenv.
+This will load a working version of Ansible.
+
+## Monitoring
+
+We use Prometheus to scrape and store timeseries
+from IPFS and the hosts themselves.
+Grafana provides the dashboard UI.
+
+Both are available at http://metrics.i.ipfs.io
+http://metrics.i.ipfs.io/prometheus, respectively.
+
+- You need to be peered into the Hyperboria cjdns network
+  in order to reach the address that this domain name points to.
+- Your cjdns address needs to be whitelisted.
+
+### Cjdns
+
+In addition to serving http://h.gateway.ipfs.io,
+we use cjdns for a very simple VPN based on an IP address whitelist.
+
+```sh
+$ git clone https://github.com/hyperboria/cjdns.git
+$ cd cjdns/
+$ ./do
+$ ./cjdroute --genconf > cjdroute.conf
+```
+
+This creates a `tunX` network interface,
+which grabs all `fc00::/8` traffic and hands it to cjdns.
+
+Scripts for various init systems are provided in `contrib/`.
+
+### Peering
+
+Cjdns nodes peer automatically on local networks,
+but on WANs like the internet, peering requires credentials.
+You can peer with the Solarnet cjdns nodes
+by using one of the `cjdns_authorized_passwords` in `secrets.yml`.
+
+Note that peering with any other node in the Hyperboria network is sufficient, too.
+
+You can check the status of your peerings:
+
+```sh
+$ watch tools/peerStats
+```
+
+Or check connectivity to solarnet:
+
+```sh
+$ ping6 h.gateway.ipfs.io
+```
+
+### Authentication
+
+Since IP packets in cjdns are authenticated and encrypted,
+we can use the HTTP client's IP address for authentication,
+instead of Basic Auth or another login mechanism.
```

Comment:
> \<cjd> larsg: answer is the crypto is able to work with unreliable messages
> \<larsg> cjd: so the cipher's local state doesn't change over time, with new nonces and all that?
> \<cjd> only when setting up the state and in that period it requires one message of type 0, then will send all further messages of type 1 until it receives one message of type 2
> \<cjd> on the other side, it will send all messages of type 0 until it receives a single message of type 1, then it will send all messages of type 2
> \<cjd> but after setup, each packet has a nonce which is a counter, autoincremented by the sender
> \<cjd> there is an anti-dupe-packet subsystem which will not allow packets which are way way *way* out of order (more then 64 packets old) but other than that, out of order and missing packets are ok for the crypto